File: HJ17.go
/*
 * 开发一个坐标计算工具， A表示向左移动，D表示向右移动，W表示向上移动，S表示向下移动。从（0,0）点开始移动，从输入字符串里面读取一些坐标
 *
 * 输入：
 *     A10;S20;W10;D30;X;A1A;B10A11;;A10;
 *
 * 输出：
 *     10,-10
 */
package main

import (
"bufio"
"fmt"
"os"
"strconv"
"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	for {
		data, _ ,err := reader.ReadLine()
		if err != nil{
			break
		}

		sList := strings.Split(string(data),";")
		var x = 0
		var y = 0

		for _, list:= range sList {
			/* 坐标移动命令长度判断 */
			if  len(list) == 2 || len(list) == 3 {
				if list[0] == 'A'{
					step, err := strconv.Atoi(list[1:])
					if err != nil {
						continue
					}
					x -= step
				}
				if list[0] == 'D'{
					step, err := strconv.Atoi(list[1:])
					if err != nil{
						continue
					}
					x += step
				}
				if list[0] == 'W'{
					step, err := strconv.Atoi(list[1:])
					if err != nil{
						continue
					}
					y += step
				}
				if list[0] == 'S'{
					step, err := strconv.Atoi(list[1:])
					if err != nil{
						continue
					}
					y -= step
				}
			}
		}
		fmt.Printf("%d,%d\n",x,y)
	}
}

File: HJ17.c
/*
 * 开发一个坐标计算工具， A表示向左移动，D表示向右移动，W表示向上移动，S表示向下移动。从（0,0）点开始移动，从输入字符串里面读取一些坐标
 *
 * 输入：
 *     A10;S20;W10;D30;X;A1A;B10A11;;A10;
 *
 * 输出：
 *     10,-10
 */

#include <stdio.h>
#include <stdlib.h>

int main(void)
{
    int count = 0;
    char in[2000] = { '\0' };
    char temp[20] = { '\0' };
    while (scanf("%s", in) != EOF)
    {
        int i = 0, x = 0, y = 0, j = 0;
        while (in[i] != '\0')
        {     
            if (in[i] != ';')
            {
                temp[j++] = in[i++];
                count++;
            }
            else
            {
                j = 0;
                i++;
                if (count == 3)
                {
                    if ((temp[j] == 'A' || temp[j] == 'D' || temp[j] == 'W' || temp[j] == 'S') &&
                        (temp[j + 1] >= '0' && temp[j + 1] <= '9') && (temp[j + 2] >= '0' && temp[j + 2] <= '9'))
                    {
                        switch (temp[j])
                        {
                        case 'A':x -= (temp[j + 1]-'0') * 10 + (temp[j + 2]-'0'); break;
                        case 'D':x += (temp[j + 1] - '0') * 10 + (temp[j + 2] - '0'); break;
                        case 'W':y += (temp[j + 1] - '0') * 10 + (temp[j + 2] - '0'); break;
                        case 'S':y -= (temp[j + 1] - '0') * 10 + (temp[j + 2] - '0'); break;
                        default:break;
                        }
                        count = 0;
                    }
                    else
                    {
                        count = 0;
                        continue;
                    }
                             
 
                }
                else if (count == 2)
                {
                    if ((temp[j] == 'A' || temp[j] == 'D' || temp[j] == 'W' || temp[j] == 'S') &&
                         (temp[j + 1] >= '0' && temp[j + 1] <= '9'))
                    {
                        switch (temp[j])
                        {
                        case 'A':x -= (temp[j + 1] - '0') ; break;
                        case 'D':x += (temp[j + 1] - '0') ; break;
                        case 'W':y += (temp[j + 1] - '0') ; break;
                        case 'S':y -= (temp[j + 1] - '0') ; break;
                        default:break; 
                        }
                        count = 0;
                    }
                    else
                    {
                        count = 0;
                        continue;
                    }            
                }
                else
                {
                    count = 0;
                    continue;
                }       
            }
        }
        printf("%d,%d\n", x, y);
    }
         
    return 0;
}